table-api/src/main/java/io/deephaven/api/filter/FilterNot.java

```diff
@@ -4,9 +4,12 @@
 package io.deephaven.api.filter;
 
 import io.deephaven.annotations.SimpleStyle;
+import io.deephaven.api.expression.Expression;
 import org.immutables.value.Value.Immutable;
 import org.immutables.value.Value.Parameter;
 
+import java.util.Optional;
+
 /**
  * Evaluates to {@code true} when the given {@link #filter() filter} evaluates to {@code false}.
  */
```

Comment:
There are reasons why I don't think its appropriate to do the simplifications at this level. There may be certain use cases where you want to preserve "nots" (up to a certain point) instead of "simplifying".

As a handwavy motivator, if we are translating the SQL expression (/filter) `NOT(NOT(NOT(Foo)))` into DHC, it's not _unreasonable_ to preserve that not logic all the way until it gets into the engine for execution instead of immediately simplifying to `NOT(Foo)`. Maybe we want to print out the QST and we want it to look as close as possible to the original users query as possible.